run_tests.sh

```diff
@@ -0,0 +1,7 @@
st,RDFDatetimeTest,FilehashTest,XoredSearchingTest,GRRFuseTest,ClientCommsTest,FilterTests,AFF4SparseImageTest,ByteSizeTest,OSXEnumerateRunningServicesTest,TimelineTest,TestTimelines,TestPackedVersionedCollection,VFSDirectoryTest,FindTest,RekallTests,RDFIntegerTest,TestExecutePython,MethodTest,EmailPluginTest,PackedVersionedCollectionCompactorTest,OSXMemoryTest,HashFileStoreTest,ArtifactFlowTest,AFF4ObjectLabelsListTest,TestExports,AuthTest,ConfigLibTest,GrepTests,GrrKbTest,TestArtifactCollectors,TriggersTest,UsersTest,SshdConfigTest,TestLinuxMemory,DurationTest,ClientURNTests,SqliteDataStoreTest,DataStoreTest,AFF4ObjectLabelTest,TestCheckFlows,TDBDataStoreBenchmarks,ThreadPoolTest,AFF4GRRTest,ChromeHistoryTest,GetClientStatsActionTest,TestSearchFileContent,TestRegistryFinderFlow,AFF4LabelsIndexTest,ListProcessesTest,Ipv6UtilsTest,RDFStructsTest,BuildTests,TestPlistFlows,FileStoreHashTest,ApiConfigRendererTest,IEHistoryTest,RunningStatsTest,AdministrativeFlowTests,YamlConfigTest,RDFValueBaseTest,RDFBytesTest,FileExportPluginTest,TestCopyPathToFile,SqliteDataStoreCSVBenchmarks,RDFValueBenchmark,TDBDataStoreTest,TestLinuxMemoryAnalysis,StoreTests,DataStoreBenchmarks,HintsTests,RekallResponseConverterTest,WindowsRegistryParserTest,ClientUtilsTest,GRRTempFileTestDirectory,EmptyActionTest,FakeDataStoreBenchmarks,AverageMicroBenchmarks,TestListVolumeShadowCopies,WindowsPersistenceMechanismsParserTest,HashDigestTest,WMIParserTest,CollectionExportPluginTest,StatsTests,MasterTest,ApiAff4RendererTest,SendEmailTests,TestClientInterrogate,TestExportHuntResultsFilesAsArchive,RDFURNTest,LinuxSoftwareParserTest,ArtifactParserTest,TestFilesystem,HuntTest,BatchConverterTest,TestArtifactCollectorsInteractions,OSXQuarantineTest,CollectorTest,CollectionFilesExportPluginTest,StandardHuntTest,UserTests,TestSearchFileContentWithFixture,TestCollections,ArtifactParserTests,DarwinPersistenceMechanismsParserTest,RDFValueCollectionApiObjectRendererTest,WebHistoryFlowTest,HTTPDataStoreTest,EmbeddedRDFValueTest,DataAgnosticExportConverterTest,GRRSeleniumTest,SqliteDataStoreBenchmarks,TestSystemRootSystemDriveFallbackFlow,OutputpluginsTest,ConfigFileTest,FlowTest,RegistryFlowTest,ExportTest,MatchMethodTests,ResultSetTest,FakeDataStoreTest,HintDefinitionTests,TestAdministrativeFlows,GRRTempFileTestFilename,FilestoreStatsCronFlowTest,ForemanTests,ReportsTest,GRRBaseTest,TestEndToEndTestFlow,DictTest,TestArtifactCollectorsRealArtifacts,KnowledgeBaseUserMergeTest,HandlerTests,NetstatTest,BasicContextTests,ArtifactHandlingTest,ObjcTest,TestWindowsMemory,WindowsActionTests,GrepTest,SystemCronFlowTest,UtilsTest,DataStoreCSVBenchmarks,CheckLoaderTests,TDBDataStoreCSVBenchmarks,ArtifactKBTest,MultiShardedQueueManagerTest,ConfigActionTest,SearchTest,RegistryVFSTests,OSXDriverTests,RekallTestSuite,GeneralFlowsTest,FlowStateTest,TestCryptoTypeInfos,AFF4Benchmark,GRRArtifactTest,GRRFuseDatastoreOnlyTest,TestOSXFileParsing,AFF4SymlinkTest,FlowTestsBaseclass,ConditionTest,RekallVADParserTest
```

Comment:
> disabling tests is not really helping here.

It's helping because you can see that this limited set of tests that is enabled is passing or failing. And the new tests that are checked in are passing or failing. You may still need to run the rest of the test suite manually to check the rest of the tests, but this provides a basic level of automation, which is useful. In particular, we can mindlessly ask external contributors to "fix the build" when we see red, without having to go into any manual labor. I think there's value in this, especially as we start welcoming additional external contributors.

> What exactly is the goal of this CL? 

The goal of this pull request is as follows:

1. To investigate the travis environment and the requirements for using travis as a build system for continuous integration. To look into its resource limits, dependencies, to time its behavior and to understand how we need to adapt to be able to run our build on it.
2. To define a clear path of what course of action will be required to achieve our quarter-goal of getting Travis fully integrated.
3. To enable the travis build system gradually in order to be able to automate some of the review process for external contributors and internal contributors who are using the PR system to submit review requests.
4. To increase the confidence in our open source build for us and for third parties in a way that is easily accessible and verifiable with a boolean indicator.
5. To make our build process more transparent to the open source community by illustrating the way tests are supposed to run.
6. To highlight the shortcomings of our current build process and to acknowledge that we are not perfect and we're willing to improve our coverage.

Just because the open source tests work for you in your machine with specific resources, specific dependencies pre-installed, and a specific version of your operating system doesn't mean they can run for everyone. One of the points of open source is to make things available to others as well, and googlers just claiming the test suite works for them is not enough. The migration to travis clearly points out the environment and configuration needed to run the tests.

> If the goal is to get travis build/testing working then it means all tests must pass.

This is not a black and white, all-or-nothing issue. All non-trivial software migration happens gradually.

> If you are saying that not all tests are passing then it means this CL is not complete and we should wait until its finished.

[Like Netscape 6.0]?

> just to have that undone in the next CL

Why would anything be undone? The next CLs will slowly improve our coverage in the way planned, until we are able to run all the tests.

> It feels sloppy and there is no reason for it other than shortness of time.

There are several reasons as indicated above. The most striking one is review rejection automation.

>  If indeed you suggest we adopt a policy of not accepting external PRs which break the build (i.e. they introduce bugs that affect existing tests) that means that those PRs are not finished either. Do you suggest we accept PRs that say - look at this new feature. I know it breaks some tests but i will fix those in the next PR. Clearly not.

No. Here's what I suggest: A PR that breaks the build should never be merged and it should be dismissed with a simple "Please fix the build" comment without further ado. A PR that passes the build should be reviewed and the reviewer should inspect the readability and semantics of the code and run the tests manually.

> It is concerning that tests are being killed due to OOM - does this mean travis as a platform is too small for us to run on?

No. The tests [are able to run just fine in single-process mode without running out of memory](https://s3.amazonaws.com/archive.travis-ci.org/jobs/46216964/log.txt). This is an indicator that travis is suitable for the open source build. Regardless of that matter, I don't think using travis is being disputed as a decision.